### PR TITLE
Lexmatch revamp

### PR DESCRIPTION
**Pull request by matentzn**
This PR 

- adds a few missing prefixes to merged.db (through prefixes.csv) that were previously missed
- replaces the `prefix_map` with `converter` parameter in `src/scripts/lexmatch-sssom-compare.py`
- skips `meta` in `src/scripts/match-mondo-sources-all-lexical.py` which is not urgently needed and was causing all the problems @joeflack4 is fixing
- amends to preprocessing queries in the pipeline that added junk xrefs to the pipeline
- Updates `src/ontology/config/prefixes.csv` to cover all relevant prefixes for the matching
- updates `src/ontology/metadata/*` configs to avoid conflicting namespaces.

@joeflack4 I am not reading your answers etc - please review this PR with @twhetzel and see if you have any concern with these changes.

To test this PR if you like:

```
docker pull obolibrary/odkfull:dev
ODK_TAG=dev ./run.sh make lexmatch/README.md -B IMP=false PAT=false MIR=false
```

I ran it locally - now testing the whole pipeline (I circumvent the main problem you are trying to solve by simply ditching `meta` here, hoping its not needed).
```

### CODE REVIEW

File: src/scripts/match-mondo-sources-all-lexical.py
Comment (joeflack4) [inline src/scripts/match-mondo-sources-all-lexical.py:89]: ## `lexical_index_to_sssom(meta=None)`
Nico [wrote](https://github.com/monarch-initiative/mondo-ingest/pull/397#issue-2094799963):
> skips `meta` in `src/scripts/match-mondo-sources-all-lexical.py` which is not urgently needed and was causing all the problems @joeflack4 is fixing

I don't know if there would be any negative side effects to this, but you are more familiar with all of the lexmatch outputs and you say that running it locally was fine, so I see no problem w/ that.

Comment (joeflack4) [inline src/scripts/match-mondo-sources-all-lexical.py:89]: ## #394 redundancy
The main difference is that #397 does passes `meta=None` to `lexical_index_to_sssom()`, whereas #394 passes a `Converter`. Supposedly #397 approach is fine.

Everything else in that PR is now cosmetic, so i can merge just those changes afterwards or ignore and close.

My instinct is to feel a little irked that I spent a lot of work and found a way to get the pipeline working without throwing the `oio` err (although #394 currently has `oio` removed and loses outputs), and that my work had I think the same overall effect but was supplanted by Nico and Charlie's PRs. But maybe I shouldn't feel irked; I laid the groundwork and this is just the nature of collaboration, and Nico/Charlie are more senior here and have a better idea of how things should exactly be.

Also, I see that this PR does add more than #394, such as missing prefixes.

File: src/ontology/config/prefixes.csv
Comment (joeflack4) [inline src/ontology/config/prefixes.csv:230]: ## `prefixes.csv` updates
That will be supplanted eventually by dynamic creation of `prefixes.csv` but these changes seem necessary for now.

Harshad and I should have thought to check the status of this sooner.

File: src/sparql/fix_xref_prefixes.ru
Comment (joeflack4) [inline src/sparql/fix_xref_prefixes.ru:33]: ## xref fixes
I don't know where these were happening but I'm glad you located and fixed!

File: src/sparql/rm_xref_by_prefix.ru
Comment (joeflack4) [inline src/sparql/rm_xref_by_prefix.ru:19]: ## URI xref removals
From what I can see, this only happens for NCIT and DOID, and it only happens after prefixes have already been renamed, so if I understand correctly, it looks like any xrefs that are no longer prefixed / recognized will be removed because they are thus in a namespace we haven't declared that we care about. So this seems fine.

File: src/scripts/lexmatch-sssom-compare.py
Comment (joeflack4) [inline src/scripts/lexmatch-sssom-compare.py:207]: ## `preifx_map` -> `converter`
This is good; allows us to use the EPM (inside of `converter`) instead.

**Review by joeflack4 — APPROVED** [review]
I trust your review of the outputs and if you say it is working locally and that you're verifying by running the pipeline, I'm fine for you / us to merge this after that verification has been made!

**Comment by twhetzel** [comment]
> I ran it locally - now testing the whole pipeline (I circumvent the main problem you are trying to solve by simply ditching meta here, hoping its not needed).

@joeflack4 have you also run the entire pipeline locally? Did you find that Nico's solution to "ditch `meta`" worked as intended?

File: src/scripts/match-mondo-sources-all-lexical.py
Comment (joeflack4) [reply src/scripts/match-mondo-sources-all-lexical.py:89]: _[Trish wrote](https://github.com/monarch-initiative/mondo-ingest/pull/397#issuecomment-1906442109)_:
> @joeflack4 have you also run the entire pipeline locally? Did you find that Nico's solution to "ditch meta" worked as intended?

I haven't run the whole pipeline locally. I'll give it a go but I'm terrified of it; I've heard it takes over 24 hours and sometimes this can interfere w/ other projects I'm working on. My computer wasn't actually capable of doing this until last month (without some skipping of things like NCIT). When I've asked Nico if he wants me to do this before, he's said I could hold off. But I'll at least give it a go.

Regarding checking myself the results of `meta=None`, I'll give that a go. I was trusting Nico and it sound like he got the results he expected, but good to have second eyes especially if you think so. Another difference I just noticed as well is that I've also been passing no `meta` (same as passing `None`), but I've been passing `prefix_map` instead. I'm not sure it matters in this case but I'd better remove that as well and run in a more closely to what Nico is running. I'd check out his PR as well but I think better to run on my PR because Nico made some other changes to `prefixes.csv`, but I want to more closely compare the outputs I got before/after with simply the `meta=None`/`prefix_map=None` change.

File: src/scripts/lexmatch-sssom-compare.py
Comment (matentzn) [reply src/scripts/lexmatch-sssom-compare.py:207]: afaics `prefix_map` does not even exist anymore in sssom-py.

File: src/ontology/mondo-ingest.Makefile
Comment (matentzn) [inline src/ontology/mondo-ingest.Makefile:345]: @joeflack4 can you make an issue about this? A clean run of `make sync` does not work because we get an error: `no goal for reports/doid.subclass.direct-in-mondo-only.tsv`. There is indeed no such goal!

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:345]: @matentzn Sorry about this. But it it strange. There is a goal, and I'm getting no such error. I created an issue though: #398

Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:345]: Ok, can you open a PR to reinstate the sync pipeline, try a fresh checkout of the repo (different folder) and see if you can replicate the problem?

File: src/ontology/mondo-ingest.Makefile
Comment (matentzn) [inline src/ontology/mondo-ingest.Makefile:433]:
```suggestion
```

**Review by matentzn — COMMENTED** [review]
Approved.. Phew!

@twhetzel @joeflack4 

I looked through all the results - They look correct to me. A lot of issues fixed, way beyond the the original problem with the prefixes. 

I wont be looking at this anymore this week - please merge this, write down any question you have in a Google docs so we can document things more permanently, and I will, once I see the merge, immediately make another run.

@joeflack4 if you want to retain the contents of your comments, please move them to issues.

This was a bit painful, sorry about that. The source of all problems, was a major refactor of `sssom-py` and `curies`, which led to 3 completely independent major issues all around the pipeline. The debugging process has led to some good fixes in `curies` (thanks @joeflack4), and I hope some better understanding of what is happening. Lets keep one thing in mind for the future:

Never _ever_ again do we merge a branch unless a full run of the pipeline was successful. I can be responsible for the runs until we have a runner on GCP.

@twhetzel I will post a comment on slack now to advice Nicole and Sabrina to push forward the syncing.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:345]: Alright

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:345]: @matentzn I'm just tried a fresh clone, running on `main`, before and after clearing `reports/` of any `sync` related outputs.

I'm not getting any errors, but no outputs either. Strange.
<details><summary>Example of just DOID</summary>
<p>

```
joeflack4@MacBook-Pro ontology % sh run.sh make sync-subclassof-doid
Running  with -Xmx20G of memory for ROBOT and Java-based pipeline steps.
make: Entering directory '/work/src/ontology'
...
Makefile:447: warning: ignoring old recipe for target 'components/omim.owl'
...
make reports/doid.subclass.direct-in-mondo-only.tsv
make[1]: Entering directory '/work/src/ontology'
...
mondo-ingest.Makefile:66: warning: overriding recipe for target 'components/omim.owl'
make pip-oaklib
make[2]: Entering directory '/work/src/ontology'
...
python3 -m pip install --upgrade oaklib
Defaulting to user installation because normal site-packages is not writeable
...
make[2]: Leaving directory '/work/src/ontology'
python3 ../scripts/sync_subclassof.py \
--outpath-added reports/doid.subclass.added.robot.tsv \
--outpath-added-obsolete reports/doid.subclass.added-obsolete.robot.tsv \
--outpath-confirmed reports/doid.subclass.confirmed.robot.tsv \
--outpath-direct-in-mondo-only reports/doid.subclass.direct-in-mondo-only.tsv \
--mondo-db-path tmp/mondo.db \
--mondo-ingest-db-path tmp/mondo-ingest.db \
--mondo-mappings-path tmp/mondo.sssom.tsv \
--onto-config-path metadata/doid.yml
/work/src/ontology/../scripts/sync_subclassof.py:47: DeprecationWarning:
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
If this would cause problems for you,
please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466

  import pandas as pd

```

</p>
</details> 

I don't have anything in my `tmp/` folder but it's still running because this isn't a fresh ODK container either.

@matentzn What ODK version are you using? I'm using `dev`.

I'm gonna try `make` locally without ODK and see what the results are.

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:345]: Ok, immediately when I tried `make` outside of ODK I got what you got:
`make[1]: *** No rule to make target `reports/doid.subclass.direct-in-mondo-only.tsv'.  Stop.`

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:345]: @matentzn Gonna move discussion to:
- #402

File: src/ontology/metadata/doid.metadata.sssom.yml
Comment (twhetzel) [inline src/ontology/metadata/doid.metadata.sssom.yml:34]: @matentzn why are these commented out vs. removed or what are these comments a reminder of in the future?

**Review by twhetzel — APPROVED** [review]
Nico mentioned all looks good and it looks good to me at this point

**Pull request merged**
